common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationSupport.java

```diff
@@ -118,7 +118,9 @@ public static boolean execBinary(final UTF8String l, final UTF8String r) {
       return l.contains(r);
     }
     public static boolean execLowercase(final UTF8String l, final UTF8String r) {
-      return l.containsInLowerCase(r);
+      if (r.numChars() == 0) return true;
+      if (l.numChars() < r.numChars()) return false;
```

Comment:
I don't think this is correct. Consider for example `l` being `İİİ` and `r` being `i̇i̇`. The result is obviously `true` but `l` has 3 characters and `r` has 4.

Reply:
let me check if this is what you mean here, considering `haystack="İİİ"` and `needle="i̇i̇"`

there exists `len>0` such that `start=0` and `len=2` satisfy the condition `lower(substring(haystack, start, len)) == lower(needle)` i.e. `lower(substring("İİİ", 0, 2)) == lower("İİ") == "i̇i̇" == lower("i̇i̇")`

hence, `startsWith("İİİ", "i̇i̇")` should actually return `true`
in addition, `contains("İİİ", "i̇i̇")` should also return `true`
and `locate("İİİ", "i̇i̇")` should return `1` (needle found at the beginning of the haystack)

Reply:
update: resolved in latest commit